plugins/kinetic-dp/fu-kinetic-dp-aux-isp.c

```diff
@@ -0,0 +1,288 @@
+	g_return_val_if_fail(br_id_str_buf_size >= DPCD_SIZE_BRANCH_DEV_ID_STR, FALSE);
+	g_return_val_if_fail(dev_info != NULL, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+	dev_info->chip_id = FU_KINETIC_DP_CHIP_NONE;
+	dev_info->fw_run_state = FU_KINETIC_DP_FW_STATE_NONE;
+
+	/* find the device info by branch ID string */
+	for (guint32 i = 0; i < num; i++) {
+		if (memcmp(br_id_str_buf,
+			   kt_dp_branch_dev_info_table[i].id_str,
+			   kt_dp_branch_dev_info_table[i].str_len) == 0) {
+			/* found the chip in the table */
+			dev_info->chip_id = kt_dp_branch_dev_info_table[i].chip_id;
+			dev_info->fw_run_state = kt_dp_branch_dev_info_table[i].fw_run_state;
+			return TRUE;
+		}
+	}
+
+	/* there might not always be nul character in the DPCD branch ID string */
+	str = g_strndup((const gchar *)br_id_str_buf, DPCD_SIZE_BRANCH_DEV_ID_STR);
+	g_set_error(error,
+		    FWUPD_ERROR,
+		    FWUPD_ERROR_INTERNAL,
+		    "%s is not a supported Kinetic device",
+		    str);
+	return FALSE;
+}
+
+gboolean
+fu_kinetic_dp_aux_isp_enable_aux_forward(FuKineticDpConnection *connection,
+					 FuKineticDpChip root_dev_chip_id,
+					 FuKineticDpFwState root_dev_state,
+					 FuKineticDpDev target_port,
+					 GError **error)
+{
+	if (root_dev_state != FU_KINETIC_DP_FW_STATE_APP) {
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_INTERNAL,
+			    "host device [%s %s] doesn't support to enable AUX forwarding",
+			    fu_kinetic_dp_aux_isp_get_chip_id_str(root_dev_chip_id),
+			    fu_kinetic_dp_fw_state_to_string(root_dev_state));
+		return FALSE;
+	}
+
+	if (root_dev_chip_id == FU_KINETIC_DP_CHIP_JAGUAR_5000 ||
+	    root_dev_chip_id == FU_KINETIC_DP_CHIP_MUSTANG_5200) {
+		if (!fu_kinetic_dp_secure_aux_isp_enable_aux_forward(connection,
+								     target_port,
+								     error)) {
+			g_prefix_error(error, "failed to enable AUX forwarding: ");
+			return FALSE;
+		}
+		/* Wait 10ms for host to process AUX forwarding command */
+		g_usleep(10 * 1000);
+		return TRUE;
+	}
+
+	g_set_error(error,
+		    FWUPD_ERROR,
+		    FWUPD_ERROR_INTERNAL,
+		    "host device [%s] doesn't support to enable AUX forwarding",
+		    fu_kinetic_dp_aux_isp_get_chip_id_str(root_dev_chip_id));
+	return FALSE;
+}
+
+gboolean
+fu_kinetic_dp_aux_isp_disable_aux_forward(FuKineticDpConnection *connection,
+					  FuKineticDpChip root_dev_chip_id,
+					  FuKineticDpFwState root_dev_state,
+					  GError **error)
+{
+	if (root_dev_state != FU_KINETIC_DP_FW_STATE_APP) {
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_INTERNAL,
+			    "host device [%s %s] doesn't support to disable AUX forwarding",
+			    fu_kinetic_dp_aux_isp_get_chip_id_str(root_dev_chip_id),
+			    fu_kinetic_dp_fw_state_to_string(root_dev_state));
+		return FALSE;
+	}
+
+	if (root_dev_chip_id == FU_KINETIC_DP_CHIP_JAGUAR_5000 ||
+	    root_dev_chip_id == FU_KINETIC_DP_CHIP_MUSTANG_5200) {
+		/* wait 5ms */
+		g_usleep(5 * 1000);
```

Comment:
can we have a `#define` for the reason used instead of just randomly sprinkling sleeps?